seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java

```diff
@@ -0,0 +1,257 @@
+                vals.add(seaTunnelRow.getFields()[partitionFieldsIndexInRow.get(i)].toString());
+            }
+            partitionDir = VariablesSubstitute.substitute(partitionDirExpression, valueMap);
+        }
+        partitionDirAndValuesMap.put(partitionDir, vals);
+        return partitionDirAndValuesMap;
+    }
+
+    /**
+     * use transaction id generate file name
+     *
+     * @param transactionId transaction id
+     * @return file name
+     */
+    @Override
+    public String generateFileName(String transactionId) {
+        String fileNameExpression = textFileSinkConfig.getFileNameExpression();
+        FileFormat fileFormat = textFileSinkConfig.getFileFormat();
+        if (StringUtils.isBlank(fileNameExpression)) {
+            return transactionId + fileFormat.getSuffix();
+        }
+        String timeFormat = textFileSinkConfig.getFileNameTimeFormat();
+        DateTimeFormatter df = DateTimeFormatter.ofPattern(timeFormat);
+        String formattedDate = df.format(ZonedDateTime.now());
+        Map<String, String> valuesMap = new HashMap<>();
+        valuesMap.put(Constants.UUID, UUID.randomUUID().toString());
+        valuesMap.put(Constants.NOW, formattedDate);
+        valuesMap.put(timeFormat, formattedDate);
+        valuesMap.put(Constant.TRANSACTION_EXPRESSION, transactionId);
+        String substitute = VariablesSubstitute.substitute(fileNameExpression, valuesMap);
+        return substitute + fileFormat.getSuffix();
+    }
+
+    /**
+     * prepare commit operation
+     *
+     * @return the file commit information
+     */
+    @Override
+    public Optional<FileCommitInfo2> prepareCommit() {
+        this.finishAndCloseFile();
+        // this.needMoveFiles will be clear when beginTransaction, so we need copy the needMoveFiles.
+        Map<String, String> commitMap = new HashMap<>(this.needMoveFiles);
+        Map<String, List<String>> copyMap = this.partitionDirAndValuesMap.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> new ArrayList<>(e.getValue())));
+        return Optional.of(new FileCommitInfo2(commitMap, copyMap, transactionDirectory));
+    }
+
+    /**
+     * abort prepare commit operation
+     */
+    @Override
+    public void abortPrepare() {
+        try {
+            FileSystemUtils.deleteFile(transactionDirectory);
+        } catch (IOException e) {
+            throw new RuntimeException("abort transaction " + this.transactionId + " error.", e);
+        }
+    }
+
+    /**
+     * when a checkpoint was triggered, snapshot the state of connector
+     *
+     * @param checkpointId checkpointId
+     * @return the list of states
+     */
+    @Override
+    public List<FileSinkState2> snapshotState(long checkpointId) {
+        ArrayList<FileSinkState2> fileState = Lists.newArrayList(new FileSinkState2(this.transactionId, this.checkpointId));
+        this.checkpointId = checkpointId;
```

Comment:
I think we need begin a new Transaction here. And do some thing like

```
this.transactionId = "T" + Constant.TRANSACTION_ID_SPLIT + jobId + Constant.TRANSACTION_ID_SPLIT + subTaskIndex + Constant.TRANSACTION_ID_SPLIT + checkpointId;
        this.transactionDir = getTransactionDir(this.transactionId);
        this.needMoveFiles = new HashMap<>();
        this.partitionDirAndValsMap = new HashMap<>();
        this.beingWrittenFile = new HashMap<>();
        this.beginTransaction(this.transactionId);
        this.checkpointId = checkpointId;
        return this.transactionId;
```

Reply:
Resolved.